server/src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -1561,4 +1503,53 @@ public int cullInvalidConsumerContentAccess(Owner owner, String... existingModes
         return updated;
     }
 
+    /**
+     * Fetches all unique system purpose attribute values set by all the consumers
+     * of the specified owner.
+     *
+     * @param owner
+     *  The owner the consumers belong to
+     * @param sysPurposeAttribute
+     *  The type of system purpose attribute needs to be fetched
+     * @return
+     *  A list of the all the distinct values of the system purpose attributes that the consumers
+     *  belonging to the specified owner have set
+     */
+    public List<String> getDistinctSyspurposeValuesByOwner(Owner owner, String sysPurposeAttribute)
```

Comment:
Since we have enums available for it, we should just use the SystemPurposeAttributeType enum instead of a string for the input here. We don't want to expose or couple the metamodel details to business logic, as can break things down the line.

Reply:
Ack. Updated the code to use `SystemPurposeAttributeType`.